### ISSUE TITLE
Deprecations: The TCA field 'CType' of table 'tt_content' uses the legacy way of defining 'items'. Please switch to associated array keys: label, value, icon, group, description.

### ISSUE THREAD

**Issue by SSFGizmo** [FriendsOfTYPO3/extension_builder#743]
> The TCA field 'CType' of table 'tt_content' uses the legacy way of defining 'items'. Please switch to associated array keys: label, value, icon, group, description.

OLD
```
/Resources/Private/CodeTemplates/Extbase/Configuration/TCA/Overrides/tt_content.phpt

// Add content element to selector list
\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTcaSelectItem(
    'tt_content',
    'CType',
    [
        '{plugin.name -> k:format.quoteString()}',
        '{extension.extensionKey}_{plugin.key}',
        '{extension.extensionKey}-plugin-{plugin.key}',
        '{extension.extensionKey}'
    ]
);

```

* https://docs.typo3.org/m/typo3/reference-coreapi/12.4/en-us/ApiOverview/ContentElements/AddingYourOwnContentElements.html#AddingCE-TCA-Overrides-tt_content

Perhaps it can help
NEW
```
/Resources/Private/CodeTemplates/Extbase/Configuration/TCA/Overrides/tt_content.phpt

// Add content element to selector list
\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTcaSelectItem(
    'tt_content',
    'CType',
    [
        'label' => '{plugin.name -> k:format.quoteString()}',
        'value' => '{extension.extensionKey}_{plugin.key}',
        'icon'  => '{extension.extensionKey}-plugin-{plugin.key}',
        'group' => '{extension.extensionKey}'
    ]
);

```

**TYPO3 Version**
v12.4

**Extension Builder Version:**
v12.0.0-beta.2

**Comment by SSFGizmo** [comment]
> The TCA field 'l10n_parent' of table 'xxx' uses the legacy way of defining 'items'. Please switch to associated array keys: label, value, icon, group, description.

OLD
```
/Resources/Private/CodeTemplates/Extbase/Configuration/TCA/tableName.phpt

        'l10n_parent' => [
            'displayCond' => 'FIELD:sys_language_uid:>:0',
            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.l18n_parent',
            'config' => [
                'type' => 'select',
                'renderType' => 'selectSingle',
                'default' => 0,
                'items' => [
                    ['', 0],

```

NEW
```
        'l10n_parent' => [
            'displayCond' => 'FIELD:sys_language_uid:>:0',
            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.l18n_parent',
            'config' => [
                'type' => 'select',
                'renderType' => 'selectSingle',
                'default' => 0,
                'items' => [
                    ['label' => '', 'value' => 0],

```

**Comment by SSFGizmo** [comment]
* https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Feature-97232-NewTCATypeDatetime.html

OLD
```
        'starttime' => [
            'exclude' => true,
            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.starttime',
            'config' => [
                'type' => 'input',
                'renderType' => 'inputDateTime',
                'eval' => 'datetime,int',
                'default' => 0,
                'behaviour' => [
                    'allowLanguageSynchronization' => true
                ]
            ],
        ],
```

NEW
```
        'starttime' => [
            'exclude' => true,
            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.starttime',
            'config' => [
                'type' => 'datetime',
                'format' => 'datetime',
                'default' => 0,
                'behaviour' => [
                    'allowLanguageSynchronization' => true
                ]
            ],
        ],
```

**Comment by SSFGizmo** [comment]
OLD
```
Resources/Private/CodeTemplates/Extbase/Configuration/TCA/tableName.phpt

        'hidden' => [
            'exclude' => true,
            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.visible',
            'config' => [
                'type' => 'check',
                'renderType' => 'checkboxToggle',
                'items' => [
                    [
                        0 => '',
                        1 => '',
                        'invertStateDisplay' => true
                    ]
                ],
            ],
        ],</f:if><f:if condition="{domainObject.addStarttimeEndtimeFields}">
```

NEW
```
  'hidden' => [
            'exclude' => true,
            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.visible',
            'config' => [
                'type' => 'check',
                'renderType' => 'checkboxToggle',
                'items' => [
                    [
                        'label' => '',
                        'invertStateDisplay' => true,
                    ],
                ],
            ],
        ],</f:if><f:if condition="{domainObject.addStarttimeEndtimeFields}">
```

**Comment by PKuhlmay** [comment]
Thank you very much. I will have a look at it.

**Comment by PKuhlmay** [comment]
fixed by #747

**Comment by DavidBruchmann** [comment]
Thanks for reporting!
